### PR TITLE
Feature/add testlibrary

### LINKED ISSUE #12: Add testlibrary

### PR DESCRIPTION
**Pull request by SpiraTecGabelM**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,3 +1,4 @@
 INLINE_RUNTIME_CHUNK=false
 VITE_API_SERVICE_BASE=BaseService
-VITE_DOMAIN="Domain"
\ No newline at end of file
+VITE_DOMAIN="Domain"
+VITE_API_PERSONAL_WORKTIME=WorktimeAccountingService/getPersonalWorktime
```
Comment (SpiraTecGabelM) [inline .env]: check von wo es kommt

File: .stylelintrc.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-  "extends": ["stylelint-config-standard", "stylelint-config-prettier"]
+  "extends": ["stylelint-config-standard"]
```
Comment (SpiraTecGabelM) [inline .stylelintrc.json:2]: check and fix

File: package.json [hunk 3]
```diff
@@ -8,40 +8,50 @@
...
     "build": "tsc && vite build",
     "preview": "vite preview",
-    "commit": "pnpm exec prettier . --write && pnpm eslint \"{**/*.ts,**/*.tsx}\""
+    "commit": "pnpm coverage && pnpm format && pnpm lint",
+    "test": "pnpx vitest run",
+    "coverage": "pnpx vitest run --coverage",
+    "eslinter": "pnpm eslint --fix \"{**/*.ts,**/*.tsx}\"",
+    "format": "pnpm exec prettier . --write",
+    "lint": "pnpm eslinter && pnpm stylelinter",
+    "stylelinter": "pnpx stylelint \"**/*.css\""
```
Comment (SpiraTecGabelM) [inline package.json]: rename to style

File: src/App.test.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
```
Comment (SpiraTecGabelM) [inline src/App.test.tsx]: remove

File: src/App.test.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import { afterEach, describe, expect, it } from 'vitest'
+import { render, screen, cleanup } from '@testing-library/react'
+import App from './App'
+
+describe('App', () => {
```
Comment (SpiraTecGabelM) [inline src/App.test.tsx]: mehr beschreibung

File: src/App.test.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  afterEach(cleanup)
```
Comment (SpiraTecGabelM) [inline src/App.test.tsx]: brauchst du nicht

File: src/App.test.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  it('should show title', () => {
+    render(<App />)
```
Comment (SpiraTecGabelM) [inline src/App.test.tsx]: const { getByTex } = render(<App />)

 expect(getByText('BlueAnt Time Tracker')).toBeInDocument()

File: src/App.tsx [hunk 8]
```diff
@@ -50,7 +50,14 @@ const App = (): JSX.Element => {
           <AppBar>
             <Text size="large">BlueAnt Time Tracker</Text>
             <Button
-              icon={darkMode ? <Moon /> : <Sun />}
+              icon={
+                darkMode ? (
+                  <Moon data-testid="moon" />
+                ) : (
+                  <Sun data-tesid="sun" />
```
Comment (SpiraTecGabelM) [inline src/App.tsx]: typo data-testid

Comment (SpiraTecGabelM) [reply src/App.tsx]: remove data-testid if possible try not to use it if not necessary

File: src/api/base/login.test.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { describe, expect, it } from 'vitest'
```
Comment (SpiraTecGabelM) [inline src/api/base/login.test.ts:1]: check and remove maybe

File: src/api/models/worktime.ts [hunk 10]
```diff
@@ -0,0 +1,47 @@
+export interface AbstractCustomField {
+  ID: string
+  context?: string
+  name?: string
+  mandatory?: boolean
+  active?: boolean
+  reportId?: string
+  validatorList?: any
```
Comment (SpiraTecGabelM) [inline src/api/models/worktime.ts]: no any

File: src/api/models/worktime.ts [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+  stringValue?: string
+}
+
+export interface CustomFieldList {
```
Comment (SpiraTecGabelM) [inline src/api/models/worktime.ts]: check this pls

File: src/api/models/worktime.ts (hunk 10 above)
Comment (SpiraTecGabelM) [reply src/api/models/worktime.ts]: check if we need this and implement

File: src/api/models/worktime.ts [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+  duration?: number
+  ticketID?: string
+  projectID?: string
+  taskID?: string
+  activityID: string
+  comment?: string
+  billable: boolean
+  reasonNotAccountableID?: string
+  iccID?: string
+  customFieldList?: CustomFieldList
```
Comment (SpiraTecGabelM) [inline src/api/models/worktime.ts]: is this an array?

File: src/api/worktime-accounting/getPersonalWorktime.ts [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+import { XMLParser } from 'fast-xml-parser'
+import {
+  type AbstractCustomField,
+  type CustomFieldList,
+  type EditWorktimeRequestParameter,
+  type WorkTime,
+} from '../models/worktime'
+import isNil from 'lodash/isNil'
+
+function mapToAbstractCustomField(item: any): AbstractCustomField {
```
Comment (SpiraTecGabelM) [inline src/api/worktime-accounting/getPersonalWorktime.ts]: change to arrow function

Comment (SpiraTecGabelM) [reply src/api/worktime-accounting/getPersonalWorktime.ts]: no any

File: src/api/worktime-accounting/getPersonalWorktime.ts [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+  return {
```
Comment (SpiraTecGabelM) [inline src/api/worktime-accounting/getPersonalWorktime.ts]: return {..item} as AbstractCustomField

File: src/pages/TimeOverview/TimeOverviewPage.tsx [hunk 15]
```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from 'react'
```
Comment (SpiraTecGabelM) [inline src/pages/TimeOverview/TimeOverviewPage.tsx]: remove react

File: src/pages/login/LoginPage.test.tsx [hunk 16]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
```
Comment (SpiraTecGabelM) [inline src/pages/login/LoginPage.test.tsx]: remove react

File: src/pages/login/LoginPage.test.tsx [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+import { afterEach, describe, expect, it } from 'vitest'
+import { render, screen, cleanup } from '@testing-library/react'
+import LoginPage from './LoginPage'
+
+describe('App', () => {
```
Comment (SpiraTecGabelM) [inline src/pages/login/LoginPage.test.tsx]: description

File: src/pages/login/LoginPage.test.tsx [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+  afterEach(cleanup)
```
Comment (SpiraTecGabelM) [inline src/pages/login/LoginPage.test.tsx]: remove

File: src/pages/login/LoginPage.test.tsx [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+  it('should show title', () => {
+    render(<LoginPage />)
```
Comment (SpiraTecGabelM) [inline src/pages/login/LoginPage.test.tsx]: improve

File: src/pages/login/LoginPage.tsx [hunk 20]
```diff
@@ -25,7 +25,7 @@ const LoginPage = (): JSX.Element => {
   }
 
   useEffect(() => {
-    if (user !== undefined || error !== undefined) {
```
Comment (SpiraTecGabelM) [inline src/pages/login/LoginPage.tsx]: error !== undefined missing

File: src/pages/login/LoginPage.tsx [hunk 21]
```diff
@@ -25,7 +25,7 @@ const LoginPage = (): JSX.Element => {
...
+    if (!isNil(user)) {
       setIsLoading(false)
```
Comment (SpiraTecGabelM) [inline src/pages/login/LoginPage.tsx]: setIsLoading(!isNil(user) || !isNil(error)) but check if correct

File: .env (hunk 1 above)
Comment (SpiraTecGabelM) [reply .env]: done

File: package.json (hunk 3 above)
Comment (SpiraTecGabelM) [reply package.json]: done

File: src/App.test.tsx (hunk 4 above)
Comment (SpiraTecGabelM) [reply src/App.test.tsx]: done

File: src/App.test.tsx (hunk 5 above)
Comment (SpiraTecGabelM) [reply src/App.test.tsx]: done

File: src/pages/login/LoginPage.test.tsx (hunk 17 above)
Comment (SpiraTecGabelM) [reply src/pages/login/LoginPage.test.tsx]: done

File: src/pages/login/LoginPage.test.tsx (hunk 18 above)
Comment (SpiraTecGabelM) [reply src/pages/login/LoginPage.test.tsx]: done

File: src/pages/login/LoginPage.test.tsx (hunk 19 above)
Comment (SpiraTecGabelM) [reply src/pages/login/LoginPage.test.tsx]: done

File: src/pages/login/LoginPage.tsx (hunk 20 above)
Comment (SpiraTecGabelM) [reply src/pages/login/LoginPage.tsx]: done

File: src/pages/login/LoginPage.tsx (hunk 21 above)
Comment (SpiraTecGabelM) [reply src/pages/login/LoginPage.tsx]: done

File: .stylelintrc.json (hunk 2 above)
Comment (SpiraTecGabelM) [reply .stylelintrc.json:2]: done

**Review by EdgarKisman — APPROVED** [review]

**Pull request merged**